public/scripts/index.js

```diff
@@ -4,6 +4,14 @@
 */
 const toggleModal = () => {
   document.querySelector(".modal").classList.toggle("modal--hidden")
+
+  let hiddenModal = document.querySelector(".modal").classList.value
+  let iFrameDiv =  document.querySelector(".modal iframe")
+  if (hiddenModal === "modal modal--hidden") {
+    iFrameDiv.setAttribute("src", '')
+  } else {
+    iFrameDiv.setAttribute("src", "https://www.youtube.com/embed/QH2-TGUlwu4")
```

Comment:
Interesting solution. I don't like having the URL duplicate on both the `iframe` and here. When the modal is hidden, we can: 1. get the `src`. 2. set the `src` to empty. 3. set the `src` to the value we retrieved in step 1.
Another option is to look at YouTube's API and see if we can pause the video rather than stop it. Can you please check if it's possible?

Reply:
Replaced the url by declaring a videoSrc const outside toggleModel function.

Reply:
still work in progress. found this link helpful <https://developers.google.com/youtube/iframe_api_reference#Getting_Started>